cs-fun-practice.rb

```diff
@@ -0,0 +1,50 @@
+# PROBLEM 2
+
+def stretch(array)
+  new_array = []
+  array.length.times do |a|
+    if array[a] % 2 == 0
+      2.times do
+        new_array << array[a] / 2
+      end
+    else
+      new_array << ((array[a] / 2) + 1)
+      new_array << array[a]/2
+    end
+  end
+  return new_array
+end
+
+
+# PROBLEM 3
+
+def numUnique(array)
+  new_hash = {count: 0}
+  if array.length == 1 || array.length == 2 && array[0] == array[1]
+    new_hash[:count] = 1
+  else
+    array.length.times do |n|
+      if array[n] != array[n-1]
+        new_hash[:count] += 1
+      end
+    end
+  end
+  return new_hash[:count]
+end
```

Comment:
yay for getting it to work! that's always the first step.

This strikes me as a little odd for a couple reasons though. 

1) the first test case - I see why you needed to do it, but it just reads a little odd because I have to read the whole code to figure out why you needed it.

2) You're basically using a hash as a single counter. So you could replace all instances of new_hash[:count] with some variable, like `total`,  and the code would work the same.

What I actually hoped you'd come up with is iterating over the array and using the hash to store the values in the array as a key and the count of those array values as the values in the hash.

e.g. If the array was [1,1,1,4,5,5] 

Create a hash that has:
{ 1 => 3, 4 => 1, 5 =>2}

and then be able to count all the key/value pairs that have a value of 1 to find the total count.

Maybe not super intuitive right now, but I guarantee you will see this pattern again in the future.